src/DocumentationViewer.js

```diff
@@ -0,0 +1,219 @@
+import GObject from "gi://GObject"
+import Gtk from "gi://Gtk";
+import Gio from "gi://Gio"
+import GLib from "gi://GLib";
+import Adw from "gi://Adw";
+import WebKit from "gi://WebKit"
+
+import resource from "./DocumentationViewer.blp";
+
+Gio._promisify(
+  WebKit.WebView.prototype,
+  "evaluate_javascript",
+  "evaluate_javascript_finish"
+);
+
+Gio._promisify(
+  Gio.File.prototype,
+  "enumerate_children_async",
+  "enumerate_children_finish",
+);
+
+Gio._promisify(
+  Gio.FileEnumerator.prototype,
+  "next_files_async",
+  "next_files_finish",
+);
+
+Gio._promisify(
+  Gio.File.prototype,
+  "load_contents_async",
+  "load_contents_finish",
+);
```

Comment:
all `Gio._promisify` go to `init.js`

Please verify that they're not there already